cmd/gardenlet/app/gardenlet.go

```diff
@@ -163,6 +163,13 @@ func run(ctx context.Context, o *Options) error {
 	}
 	kubernetes.UseCachedRuntimeClients = gardenletfeatures.FeatureGate.Enabled(features.CachedRuntimeClients)
 
+	if gardenletfeatures.FeatureGate.Enabled(features.ReversedVPN) &&
+		(!gardenletfeatures.FeatureGate.Enabled(features.APIServerSNI) ||
+			gardenletfeatures.FeatureGate.Enabled(features.KonnectivityTunnel)) {
+		panic(fmt.Errorf(fmt.Sprintf("Inconsistent feature gate, APIServerSNI is required for ReversedVPN! (APIServerSNI: %t, ReversedVPN: %t)\nReversedVPN is not compatible with KonnectivityTunnel!(KonnectivityTunnel: %t)",
+			gardenletfeatures.FeatureGate.Enabled(features.APIServerSNI), gardenletfeatures.FeatureGate.Enabled(features.ReversedVPN), gardenletfeatures.FeatureGate.Enabled(features.KonnectivityTunnel))))
```

Comment:
You can simply return err, it will again lead to err on gardenlet start -> so the end result would be the same you want. In general explicit panic would make impossible the given func to be unit tested, but when you return err, you can easily unit test it.

```suggestion
		return fmt.Errorf("inconsistent feature gate: APIServerSNI is required for ReversedVPN (APIServerSNI: %t, ReversedVPN: %t) and ReversedVPN is not compatible with KonnectivityTunnel (KonnectivityTunnel: %t)",
			gardenletfeatures.FeatureGate.Enabled(features.APIServerSNI), gardenletfeatures.FeatureGate.Enabled(features.ReversedVPN), gardenletfeatures.FeatureGate.Enabled(features.KonnectivityTunnel))
```

Reply:
> In general explicit panic would make impossible the given func to be unit tested, but when you return err, you can easily unit test it.

You can test it easily https://onsi.github.io/gomega/#asserting-on-panics

On the other side  - you should probably return error here.

Reply:
Yeah, it might be possible at the end with some hacks, but still it is odd to panic in a func when all other error handlings return the err. So I would definitely vote for returning the err.

Reply:
Switched to error propagation instead of panic in latest commit.